crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1373,14 +1373,21 @@ impl<'src> Lexer<'src> {
         }
 
         let mut current_position = self.current_range().start();
-        let reverse_chars = self.source[..current_position.to_usize()].chars().rev();
+        let mut reverse_chars = self.source[..current_position.to_usize()]
+            .chars()
+            .rev()
+            .peekable();
         let mut newline_position = None;
 
-        for ch in reverse_chars {
+        while let Some(ch) = reverse_chars.next() {
             if is_python_whitespace(ch) {
                 current_position -= ch.text_len();
             } else if matches!(ch, '\n' | '\r') {
                 current_position -= ch.text_len();
+                if let Some(continuation_char) = reverse_chars.next_if_eq(&'\\') {
```

Comment:
Hmm, it might be more complicated than that... The continuation character could be escaped. But maybe that's not relevant because we already assume that we aren't inside a string?

Reply:
Good point, I'd need to account for the fact that the line continuation character itself can be escaped as well.